### PR TITLE
📖 Project Pythia Blog

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

File: docs/blog/2024-06-14-project-pythia-mystmd.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+One of the most exciting aspects of the transition was that it was almost painless, we have gone to great lengths to ensure that content is supported and backwards compatible, and there were almost no content changes required. The transition of a cookbook was almost as easy as:
+
+```bash
+conda install mystmd
+myst
+```
+
+Over the coming months we will roll out targeted upgrade routines for JupyterBook that understand and upgrade the `_config.yml`, but where we were this week was that required manual effort by the team. It was pleasantly surprising to see that this also fixed some issues out of the box!
+
+:::{figure} ./images/pythia-jb-myst.png
```
Comment (jnywong) [inline docs/blog/2024-06-14-project-pythia-mystmd.md:43]: Here's a later version of this screenshot I posted after we fixed the Project Pythia logo:

Screenshot 2024-06-12 at 14 30 18

File: docs/blog/2024-06-14-project-pythia-mystmd.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Project Pythia serves as an education and training hub for the geoscientific Python community, acting as the educational wing of the Pangeo initiative. It provides an extensive range of Python-centered learning resources that are open-source, community-owned, geoscience-focused, and of exceptional quality. These resources and tutorials or “Cookbooks” offer ways to onboard into the geoscientific python ecosystem and demonstrate ways to process and understand vast amounts of numerical scientific data using tools that support open, reproducible science, while fostering an inclusive community of practice. The Project Pythia technical stack has been based on Jupyter Book, whose next release is undergoing significant development at the moment through the mystmd project. Members of the MyST team attended the hackathon and helped with onboarding, development of new features, and bug fixes throughout the week. It was an exciting event that showcased the power of the new tools that we aim to capture in this post.
```
Comment (jnywong) [inline docs/blog/2024-06-14-project-pythia-mystmd.md:11]:
```suggestion
Project Pythia serves as an education and training hub for the geoscientific Python community, acting as the educational wing of the [Pangeo](https://pangeo.io/) initiative. It provides an extensive range of Python-centered learning resources that are open-source, community-owned, geoscience-focused, and of exceptional quality. These resources and tutorials or “Cookbooks” offer ways to onboard into the geoscientific Python ecosystem and demonstrate ways to process and understand vast amounts of numerical scientific data using tools that support open, reproducible science, while fostering an inclusive community of practice. The Project Pythia technical stack has been based on Jupyter Book, whose next release is undergoing significant development at the moment through the mystmd project. Members of the MyST team attended the hackathon and helped with onboarding, development of new features, and bug fixes throughout the week. It was an exciting event that showcased the power of the new tools that we aim to capture in this post.
```

File: docs/blog/2024-06-14-project-pythia-mystmd.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+## Cookbook Architecture
+
+The Project Pythia architecture relies on a number of Jupyter Books in separate repositories under the “Project Pythia” GitHub organization. Each Cookbook is contributed to by different authors who bring their own approaches to collaboration. To elevate the resources and incorporate them into the Pythia brand, there is a custom theme (including headers, logos and footers) and these resources are brought together in a gallery where users can filter and sort the cookbooks. In addition, there is a [Foundations cookbook](https://foundations.projectpythia.org/landing-page.html) that lays out general practices and resources that are useful throughout the computational geoscience community, with tutorials on everything from Jupyter and GitHub to the specific tools that are commonly used.
+The Project Pythia infrastructure team has faced several challenges in maintaining their infrastructure that has hindered scalability of the project as well as maintenance burden on the team:
```
Comment (jnywong) [inline docs/blog/2024-06-14-project-pythia-mystmd.md:16]:
```suggestion
The Project Pythia infrastructure team has faced several challenges in maintaining their infrastructure that has hindered the scalability of the project as well as the maintenance burden on the team:
```

File: docs/blog/2024-06-14-project-pythia-mystmd.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+Limited use of MyST in Cookbooks
+: Currently there is limited use of any advanced features of MyST Markdown in the cookbook tutorials (e.g. admonitions, cross-references, etc.) as the main way to experience a cookbook is through JupyterLab, and the default renderer of Jupyter doesn’t interpret MyST Markdown (but [jupyterlab-myst](https://github.com/executablebooks/jupyterlab-myst) does!).
+
+Throughout the week, the core ExecutableBooks team, [Curvenote](https://curvenote.com), and [2i2c](https://2i2c.org) helped address some of these constraints with the new mystmd tooling that we have been developing over the last several years.
+
+## Architecture and Workflow
+
+One of the most exciting aspects of the transition was that it was almost painless, we have gone to great lengths to ensure that content is supported and backwards compatible, and there were almost no content changes required. The transition of a cookbook was almost as easy as:
```
Comment (jnywong) [inline docs/blog/2024-06-14-project-pythia-mystmd.md:34]: Reading this again, this still blows my mind 😄

**Review by jnywong — COMMENTED** [review]
I've just had a quick glance and left a few small comments. Great blog post, you really captured the excitement of the event!

File: docs/blog/2024-06-14-project-pythia-mystmd.md (hunk 4 above)
Comment (rowanc1) [reply docs/blog/2024-06-14-project-pythia-mystmd.md:34]: 🚀

File: docs/blog/2024-06-14-project-pythia-mystmd.md (hunk 1 above)
Comment (rowanc1) [reply docs/blog/2024-06-14-project-pythia-mystmd.md:43]: Thank you! Updated!

**Comment by rowanc1** [comment]
Had a quick chat with @choldgraf and he suggested that we put a small funding statement at the bottom of the blog that mentions the pythia grant as well as Curvenote's investment in the week.

@choldgraf if you wanted to take a first pass at the, go for it. I can take that over the line tomorrow if you don't get it done.

**Comment by choldgraf** [comment]
To try and move this forward - I suggest that if we haven't got resolution on this by EOD tomorrow, we just merge the blog post as-is and we can always dig into the specific contributions in org-specific blog posts that cross-reference this one.

**Review by choldgraf — DISMISSED** [review]
I'll approve this to make it clear that I think it's good to go regardless of the changes I suggested.

To try and move this forward - I suggest that if we haven't got resolution on this by EOD tomorrow (or whenever Rowan decides, assuming nobody asks us to hold up), we just merge the blog post as-is and we can always dig into the specific contributions in org-specific blog posts that cross-reference this one.

**Comment by choldgraf** [comment]
Just noting that some images seem to be not rendering in the preview

CleanShot 2024-06-17 at 14 30 25@2x

**Comment by rowanc1** [comment]
Yes, videos will work on merge to main (jupyter-book make videos hard)!

I will write a quick funding blurb and ping you by tomorrow.

**Comment by brian-rose** [comment]
I like the suggestion to acknowledge funding and contributions.

There are three arms of the collaborative Pythia NSF award, and all three orgs made substantial contributions to last week's event:
- University at Albany (NSF award 2324302): Led the funding acquisition, helped organize and facilitate the event.
- UCAR (NSF award 2324303): Led the planning and logistics for the event.
- 2i2c / Code for Science and Society (NSF award 2324304): Provided tailored compute services and on-site support.

**Comment by jnywong** [comment]
Hey all! The [2i2c blog post](https://2i2c.org/blog/2024/project-pythia-cookoff/) is live ✨ 

I've made a cross-link to this EB blog post, so ping me when this PR is merged and I can get that cross-link updated.

**Comment by rowanc1** [comment]
@jnywong the blog on EBP side should be live shortly, good to see more perspectives on the event written down. Was a special week. :)

@brian-rose, can you ping us when there is a Pythia blog live, and we can cross-link as well!

**Pull request merged**